File: app/src/main/java/br/com/menuvem/lojista/presentation/shopping/ShoppingListsScreen.kt
package br.com.menuvem.lojista.presentation.shopping

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material.icons.outlined.List
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import br.com.menuvem.lojista.domain.model.ListaCompras
import br.com.menuvem.lojista.domain.model.StatusLista
import br.com.menuvem.lojista.presentation.components.EmptyState
import br.com.menuvem.lojista.presentation.components.formatarMoeda
import br.com.menuvem.lojista.ui.theme.PurplePrimary
import br.com.menuvem.lojista.ui.theme.SuccessGreen
import br.com.menuvem.lojista.ui.theme.YellowSecondary
import java.time.format.DateTimeFormatter

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ShoppingListsScreen(
    onNavigateToLista: (Long) -> Unit,
    onBack: () -> Unit,
    viewModel: ShoppingListsViewModel = hiltViewModel()
) {
    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
    var showCreateDialog by remember { mutableStateOf(false) }
    var newListName by remember { mutableStateOf("") }

    if (showCreateDialog) {
        AlertDialog(
            onDismissRequest = { showCreateDialog = false; newListName = "" },
            icon = { Icon(Icons.Default.Add, contentDescription = null) },
            title = { Text("Nova lista de compras") },
            text = {
                OutlinedTextField(
                    value = newListName,
                    onValueChange = { newListName = it },
                    label = { Text("Nome da lista") },
                    placeholder = { Text("Ex: Mercado - Semana 33") },
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth()
                )
            },
            confirmButton = {
                Button(
                    onClick = {
                        if (newListName.isNotBlank()) {
                            viewModel.createLista(newListName) { id ->
                                showCreateDialog = false
                                newListName = ""
                                onNavigateToLista(id)
                            }
                        }
                    },
                    enabled = newListName.isNotBlank()
                ) { Text("Criar") }
            },
            dismissButton = {
                TextButton(onClick = { showCreateDialog = false; newListName = "" }) {
                    Text("Cancelar")
                }
            }
        )
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Minhas Listas", fontWeight = FontWeight.SemiBold) },
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(Icons.Default.ArrowBack, contentDescription = "Voltar")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.surface
                )
            )
        },
        floatingActionButton = {
            ExtendedFloatingActionButton(
                onClick = { showCreateDialog = true },
                icon = { Icon(Icons.Default.Add, contentDescription = null) },
                text = { Text("Nova Lista") },
                containerColor = YellowSecondary,
                contentColor = Color.White
            )
        }
    ) { paddingValues ->
        if (uiState.isLoading) {
            Box(
                modifier = Modifier.fillMaxSize().padding(paddingValues),
                contentAlignment = Alignment.Center
            ) { CircularProgressIndicator(color = PurplePrimary) }
        } else if (uiState.listas.isEmpty()) {
            Box(
                modifier = Modifier.fillMaxSize().padding(paddingValues),
                contentAlignment = Alignment.Center
            ) {
                EmptyState(
                    icon = Icons.Outlined.List,
                    title = "Nenhuma lista criada",
                    subtitle = "Crie sua primeira lista para começar a registrar compras"
                )
            }
        } else {
            LazyColumn(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(paddingValues),
                contentPadding = PaddingValues(
                    start = 12.dp, top = 8.dp, end = 12.dp, bottom = 80.dp
                ),
                verticalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                items(uiState.listas) { lista ->
                    ListaComprasItem(
                        lista = lista,
                        onClick = { onNavigateToLista(lista.id) }
                    )
                }
            }
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
private fun ListaComprasItem(
    lista: ListaCompras,
    onClick: () -> Unit
) {
    val isFinalized = lista.status == StatusLista.FINALIZADA
    val formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy")

    ElevatedCard(
        onClick = onClick,
        modifier = Modifier.fillMaxWidth(),
        elevation = CardDefaults.elevatedCardElevation(defaultElevation = 2.dp)
    ) {
        Row(
            modifier = Modifier.padding(16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Surface(
                shape = MaterialTheme.shapes.medium,
                color = if (isFinalized)
                    SuccessGreen.copy(alpha = 0.12f)
                else
                    PurplePrimary.copy(alpha = 0.10f),
                modifier = Modifier.size(44.dp)
            ) {
                Box(contentAlignment = Alignment.Center) {
                    Icon(
                        imageVector = if (isFinalized) Icons.Default.CheckCircle else Icons.Default.ShoppingCart,
                        contentDescription = null,
                        tint = if (isFinalized) SuccessGreen else PurplePrimary,
                        modifier = Modifier.size(24.dp)
                    )
                }
            }

            Spacer(modifier = Modifier.width(12.dp))

            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = lista.nome,
                    style = MaterialTheme.typography.titleSmall,
                    fontWeight = FontWeight.SemiBold,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis
                )
                Text(
                    text = lista.dataCriacao.format(formatter),
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }

            Column(horizontalAlignment = Alignment.End) {
                if (lista.totalGasto > 0) {
                    Text(
                        text = "R$ ${formatarMoeda(lista.totalGasto)}",
                        style = MaterialTheme.typography.titleSmall,
                        fontWeight = FontWeight.Bold,
                        color = if (isFinalized) SuccessGreen else PurplePrimary
                    )
                }
                Spacer(modifier = Modifier.height(2.dp))
                Surface(
                    shape = MaterialTheme.shapes.extraSmall,
                    color = if (isFinalized)
                        SuccessGreen.copy(alpha = 0.12f)
                    else
                        PurplePrimary.copy(alpha = 0.10f)
                ) {
                    Text(
                        text = if (isFinalized) "Finalizada" else "Aberta",
                        style = MaterialTheme.typography.labelSmall,
                        color = if (isFinalized) SuccessGreen else PurplePrimary,
                        modifier = Modifier.padding(horizontal = 6.dp, vertical = 2.dp)
                    )
                }
            }

            Spacer(modifier = Modifier.width(8.dp))
            Icon(
                Icons.Default.ChevronRight,
                contentDescription = null,
                tint = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.5f)
            )
        }
    }
}

File: lib/presentation/shopping/shopping_lists_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';
import 'package:intl/intl.dart';

import '../../domain/model/lista_compras.dart';
import '../../domain/usecase/lista_compras_usecases.dart' as usecase;
import '../../providers/repository_providers.dart';
import '../../theme/app_theme.dart';
import '../components/empty_state.dart';
import '../components/formatters.dart';

final listasComprasStreamProvider = StreamProvider<List<ListaCompras>>((ref) {
  return ref.watch(listaComprasRepositoryProvider).getAllListas();
});

class ShoppingListsScreen extends ConsumerWidget {
  const ShoppingListsScreen({super.key});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final listasAsync = ref.watch(listasComprasStreamProvider);

    return Scaffold(
      appBar: AppBar(title: const Text('Minhas Listas')),
      floatingActionButton: FloatingActionButton.extended(
        onPressed: () => _showCreateDialog(context, ref),
        backgroundColor: yellowSecondary,
        foregroundColor: Colors.white,
        icon: const Icon(Icons.add),
        label: const Text('Nova Lista'),
      ),
      body: listasAsync.when(
        loading: () => const Center(child: CircularProgressIndicator(color: purplePrimary)),
        error: (e, _) => Center(child: Text('Erro: $e')),
        data: (listas) {
          if (listas.isEmpty) {
            return const EmptyState(
              icon: Icons.list_outlined,
              title: 'Nenhuma lista criada',
              subtitle: 'Crie sua primeira lista para começar a registrar compras',
            );
          }
          return ListView.separated(
            padding: const EdgeInsets.fromLTRB(12, 8, 12, 80),
            itemCount: listas.length,
            separatorBuilder: (_, __) => const SizedBox(height: 8),
            itemBuilder: (context, i) => _ListaComprasItem(
              lista: listas[i],
              onTap: () => context.push('/shopping-lists/${listas[i].id}'),
            ),
          );
        },
      ),
    );
  }

  void _showCreateDialog(BuildContext context, WidgetRef ref) {
    final controller = TextEditingController();
    showDialog<void>(
      context: context,
      builder: (context) => StatefulBuilder(
        builder: (context, setState) => AlertDialog(
          icon: const Icon(Icons.add),
          title: const Text('Nova lista de compras'),
          content: TextField(
            controller: controller,
            decoration: const InputDecoration(
              labelText: 'Nome da lista',
              hintText: 'Ex: Mercado - Semana 33',
            ),
            onChanged: (_) => setState(() {}),
          ),
          actions: [
            TextButton(
              onPressed: () => Navigator.of(context).pop(),
              child: const Text('Cancelar'),
            ),
            FilledButton(
              onPressed: controller.text.trim().isEmpty
                  ? null
                  : () async {
                      final id = await usecase.createListaCompras(
                        ref.read(listaComprasRepositoryProvider),
                        controller.text,
                      );
                      if (context.mounted) {
                        Navigator.of(context).pop();
                        context.push('/shopping-lists/$id');
                      }
                    },
              child: const Text('Criar'),
            ),
          ],
        ),
      ),
    );
  }
}

class _ListaComprasItem extends StatelessWidget {
  const _ListaComprasItem({required this.lista, required this.onTap});
  final ListaCompras lista;
  final VoidCallback onTap;

  @override
  Widget build(BuildContext context) {
    final isFinalized = lista.status == StatusLista.finalizada;
    final cor = isFinalized ? successGreen : purplePrimary;
    final formatter = DateFormat('dd/MM/yyyy');

    return Card(
      child: InkWell(
        onTap: onTap,
        child: Padding(
          padding: const EdgeInsets.all(16),
          child: Row(
            children: [
              Container(
                width: 44,
                height: 44,
                decoration: BoxDecoration(
                  color: cor.withValues(alpha: isFinalized ? 0.12 : 0.10),
                  borderRadius: BorderRadius.circular(12),
                ),
                child: Icon(
                  isFinalized ? Icons.check_circle : Icons.shopping_cart,
                  color: cor,
                ),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      lista.nome,
                      maxLines: 1,
                      overflow: TextOverflow.ellipsis,
                      style: Theme.of(context)
                          .textTheme
                          .titleSmall
                          ?.copyWith(fontWeight: FontWeight.w600),
                    ),
                    Text(
                      formatter.format(lista.dataCriacao),
                      style: Theme.of(context).textTheme.bodySmall?.copyWith(
                            color: Theme.of(context).colorScheme.onSurfaceVariant,
                          ),
                    ),
                  ],
                ),
              ),
              Column(
                crossAxisAlignment: CrossAxisAlignment.end,
                children: [
                  if (lista.totalGasto > 0)
                    Text(
                      'R\$ ${formatarMoeda(lista.totalGasto)}',
                      style: Theme.of(context).textTheme.titleSmall?.copyWith(
                            fontWeight: FontWeight.bold,
                            color: cor,
                          ),
                    ),
                  const SizedBox(height: 2),
                  Container(
                    padding: const EdgeInsets.symmetric(horizontal: 6, vertical: 2),
                    decoration: BoxDecoration(
                      color: cor.withValues(alpha: isFinalized ? 0.12 : 0.10),
                      borderRadius: BorderRadius.circular(4),
                    ),
                    child: Text(
                      isFinalized ? 'Finalizada' : 'Aberta',
                      style: Theme.of(context).textTheme.labelSmall?.copyWith(color: cor),
                    ),
                  ),
                ],
              ),
              const SizedBox(width: 8),
              Icon(
                Icons.chevron_right,
                color: Theme.of(context).colorScheme.onSurfaceVariant.withValues(alpha: 0.5),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
